cmd/tink-server/cmd/root.go

```diff
@@ -0,0 +1,247 @@
+			if err != nil {
+				logger.Fatal(err)
+			}
+			if numAvailableMigrations != 0 {
+				logger.Info("Your database schema is not up to date. Please apply migrations running tink-server with env var ONLY_MIGRATION set.")
+			}
+
+			tlsCert, certPEM, modT, err := getCerts(caCertFile, tlsCertFile, tlsKeyFile)
+			if err != nil {
+				logger.Fatal(err)
+			}
+
+			rpcServer.SetupGRPC(ctx, logger, facility, tinkDB, certPEM, tlsCert, modT, errCh)
+			httpServer.SetupHTTP(ctx, logger, certPEM, modT, errCh)
+
+			sigs := make(chan os.Signal, 1)
+			signal.Notify(sigs, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
+
+			select {
+			case err = <-errCh:
+				logger.Fatal(err)
+			case sig := <-sigs:
+				logger.With("signal", sig.String()).Info("signal received, stopping servers")
+			}
+			closer()
+
+			// wait for grpc server to shutdown
+			err = <-errCh
+			if err != nil {
+				logger.Fatal(err)
+			}
+			err = <-errCh
+			if err != nil {
+				logger.Fatal(err)
+			}
+		},
+	}
+
+	must := func(err error) {
+		if err != nil {
+			logger.Fatal(err)
+		}
+	}
+
+	rootCmd.Flags().String("facility", "", "Facility")
+
+	rootCmd.Flags().String("ca-cert", "", "File containing the ca certificate")
+
+	rootCmd.Flags().String("tls-cert", "bundle.pem", "File containing the tls certificate")
+	must(rootCmd.MarkFlagRequired("tls-cert"))
+
+	rootCmd.Flags().String("tls-key", "server-key.pem", "File containing the tls private key")
+	must(rootCmd.MarkFlagRequired("tls-cert"))
+
+	rootCmd.Flags().Bool("only-migration", false, "only run database migrations")
+
+	return rootCmd
+}
+
+// createViper creates a Viper object configured to read in configuration files
+// (from various paths with content type specific filename extensions) and loads
+// environment variables.
+func createViper(logger log.Logger) (*viper.Viper, error) {
+	v := viper.New()
+	v.AutomaticEnv()
+	v.SetConfigName("tink-server")
+	v.AddConfigPath("/etc/tinkerbell")
+	v.AddConfigPath(".")
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+
+	// If a config file is found, read it in.
+	if err := v.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			logger.With("configFile", v.ConfigFileUsed()).Error(err, "could not load config file")
+
+			return nil, err
+		}
+
+		logger.Info("no config file found")
+	} else {
+		logger.With("configFile", v.ConfigFileUsed()).Info("loaded config file")
+	}
+
+	return v, nil
+}
+
+func applyViper(v *viper.Viper, cmd *cobra.Command) error {
+	errors := []error{}
+
+	cmd.Flags().VisitAll(func(f *pflag.Flag) {
+		if !f.Changed && v.IsSet(f.Name) {
+			val := v.Get(f.Name)
+			if err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
+				errors = append(errors, err)
+
+				return
+			}
+		}
+	})
+
+	if len(errors) > 0 {
+		errs := []string{}
+		for _, err := range errors {
+			errs = append(errs, err.Error())
+		}
+
+		return fmt.Errorf(strings.Join(errs, ", "))
+	}
+
+	return nil
+}
```

Comment:
@displague I modeled this after the work you did for tink-worker, I'm thinking we can move this to a shared library that all cli tools import.